client/my-sites/plugins/search-box-header/index.jsx

```diff
@@ -29,8 +30,29 @@ const SearchBox = ( { isMobile, doSearch, searchTerm } ) => {
 
 const PopularSearches = ( props ) => {
 	const { searchTerms, siteSlug } = props;
+	const dispatch = useDispatch();
 	const translate = useTranslate();
 
+	const trackImpression = ( searchTerm ) => {
+		dispatch(
+			recordTracksEvent( 'calypso_plugins_popular_searches_impression', {
+				search_term: searchTerm,
+			} )
+		);
+	};
+
+	const onClick = ( searchTerm ) => {
+		dispatch(
+			recordTracksEvent( 'calypso_plugins_popular_searches_click', {
+				search_term: searchTerm,
+			} )
+		);
+
+		page( `/plugins/${ siteSlug || '' }?s=${ searchTerm }` );
+	};
+
+	searchTerms.forEach( trackImpression );
```

Comment:
This is generating impressions for each term, every click / search, is this too much or are we ok with sending this amount of events?

Reply:
@Rolilink Would tracking each click be enough?

Reply:
@allilevine I think so can totally remove them! also maybe a `useEffect` with an array dependency can do the trick? the impressions will only go out when the user first goes into the search then when clicking it will not rer-ender since we will pass the `searcheTerms` array as the dependency array